src/loader/x86_64/elf/mod.rs

```diff
@@ -343,15 +343,28 @@ where
         }
 
         // Skip the note header plus the size of its fields (with alignment).
-        let namesz_aligned = align_up(u64::from(nhdr.n_namesz), n_align)?;
-        let descsz_aligned = align_up(u64::from(nhdr.n_descsz), n_align)?;
+        // How to compute next notes fields can be found in this man page:
+        // https://man7.org/linux/man-pages/man5/elf.5.html.
+        let namesz_aligned = align_up(u64::from(nhdr.n_namesz), 4)?;
```

Comment:
I've done some research and my understanding is that `p_align` refers to segments alignment rather than `name` or `desc` notes fields alignment. There are also other resources that point to 4-byte alignment [1], [2], [3], [4]. [1] mentions about an alignment dependent on the arch reg size (4 bytes vs 8 bytes), so I think we lack this sort of logic as well. [5] is an old mail thread regarding how should notes be aligned on 64bit systems and the overall idea was that most applications (gdb, gcc and other binutils tools) and the linux kernel expect 4 bytes alignment even on 64bit systems. 

I suggest having only the alignment adjustment, letting arch dependent implementation arise as a requirement, if needed (it is unlikely, low priority at the moment and it is also dependent on the gABI implementation of the build toolchain used).

[1] https://refspecs.linuxbase.org/elf/gabi4+/ch5.pheader.html#note_section - LSB details on ELF notes
[2] http://www.sco.com/developers/devspecs/gabi41.pdf - Generic ABI, page 81
[3] http://lkml.iu.edu/hypermail/linux/kernel/0611.1/0538.html - discussion around modifying note alignment on 64 bit platforms.
[4] linux kernel elf core dumper `notesize` - https://elixir.bootlin.com/linux/latest/source/fs/binfmt_elf.c#L1429